.devcontainer/Dockerfile

```diff
@@ -10,9 +10,16 @@ RUN apt-get update -qq && export DEBIAN_FRONTEND=noninteractive && \
     apt-get install -y --no-install-recommends \
         software-properties-common wget apt-utils file zip \
         openssh-client gpg-agent socat rsync \
-        make ninja-build git \
+        make ninja-build git
+
+# Install python
+RUN apt-get update -qq && export DEBIAN_FRONTEND=noninteractive && \
+    apt-get install -y --no-install-recommends \
```

Comment:
Why is this being split up into a separate step?

Keep in mind, every time you run `apt-get update` you're downloading a lot of files that you just downloaded.

Reply:
I think that the downloads are cashed not sure 100% though. Also, each command should be self-contained if you change something some months later you want the most resent package to be installed right? Setting all python packages together seems more natural to me dunno

Reply:
`apt-get update` performs a package index sync. When you run it, you are telling `apt-get` that your package index is stale, and you want it to re-download the package index. There is a cache involved here, but running `apt-get update` invalidates it.

Note that https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run recommends running `apt-get update` before every `apt-get install` in a dockerfile, for the reasons you mention.

I don't see why we need the absolute latest versions of Python. We need Python here so we can use pip and Conan as package managers. These tools don't really care what version of Python you have, as long as it's 3.5+ (see Conan docs at https://docs.conan.io/en/1.43/installation.html#install-with-pip-recommended).

You picked out Python as the package you want the latest version of, and left these packages as is: `software-properties-common wget apt-utils file zip openssh-client gpg-agent socat rsync make ninja-build git`. What makes Python so different?

Reply:
I didn't get why the other packages are not the latest ones? They also have an update command before the installation take place

Reply:
If you don't have a reason for making this change, please put `python3` and `python3-pip` back in the first call to `apt-get install`, where they were before. Making this change will slow down `docker build` without providing a useful benefit.